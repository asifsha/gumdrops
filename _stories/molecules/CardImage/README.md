The `<CardImage>` component is used to append images inside of a `<Card>`. It can add an image to the top or bottom of the card.
