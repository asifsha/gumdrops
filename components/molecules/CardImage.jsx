import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const CardImage = ({ url, option, size, className, style }) => {
    const baseClass = 'gds-card__img-container';
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${option}`]: option,
        [`${baseClass}--${size}`]: size
    });

    return (
        <div className={rootClass} style={style}>
            <img className="gds-card__img" src={url} />
            <div className="gds-card__img-helper" />
        </div>
    );
};

CardImage.displayName = 'CardImage';

CardImage.defaultProps = {
    option: 'top'
};

CardImage.propTypes = {
    url: PropTypes.string.isRequired,
    /** top, bottom */
    option: PropTypes.oneOf(['top', 'bottom']),
    /** xs, sm, md, lg, xl */
    size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
    className: PropTypes.string,
    style: PropTypes.object
};

export default CardImage;
