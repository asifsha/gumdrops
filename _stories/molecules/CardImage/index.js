import React from 'react';
import { text, object } from '@storybook/addon-knobs';
import { optionalSelect } from '../../../components/utils/optionalSelect';

import readme from './README.md';
import Card from '../../../components/molecules/Card';
import CardImage from '../../../components/molecules/CardImage';

const options = {
    top: 'top',
    bottom: 'bottom',
    'No Value': ''
};

const sizeOptions = {
    'No Value': '',
    xs: 'xs',
    sm: 'sm',
    md: 'md',
    lg: 'lg',
    xl: 'xl'
};

const component = () => (
    <Card>
        <CardImage
            url={text(
                'Image URL',
                'https://c.gumgum.com/ads/com/gumgum/mantii/internal_mobile_inscreen_test/full_canvas/01/mantii_fullscreen.hyperesources/girl@2x.png'
            )}
            option={optionalSelect('Option', options, '')}
            size={optionalSelect('Size', sizeOptions, '')}
            className={text('ClassName', '')}
            style={object('Style', {})}
        />
    </Card>
);

export default [readme, component];
