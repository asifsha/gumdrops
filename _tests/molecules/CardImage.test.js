/* globals mount */
import React from 'react';
import CardImage from '../../components/molecules/CardImage';

const defaultProps = {
    option: 'top',
    className: 'custom',
    size: 'xl',
    style: { width: 120 },
    url: 'https://foo.bar'
};

describe('Expect <CardImage>', () => {
    it('to render', () => {
        const wrapper = mount(<CardImage {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});
