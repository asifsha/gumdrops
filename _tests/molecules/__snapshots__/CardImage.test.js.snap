// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <CardImage> to render 1`] = `
<CardImage
  className="custom"
  option="top"
  size="xl"
  style={
    Object {
      "width": 120,
    }
  }
  url="https://foo.bar"
>
  <div
    className="gds-card__img-container custom gds-card__img-container--top gds-card__img-container--xl"
    style={
      Object {
        "width": 120,
      }
    }
  >
    <img
      className="gds-card__img"
      src="https://foo.bar"
    />
    <div
      className="gds-card__img-helper"
    />
  </div>
</CardImage>
`;
